src/app/modules/card-cluster/card-cluster.component.ts

```diff
@@ -94,6 +94,20 @@ export class AppCardClusterComponent implements OnInit {
       }
     }
     this.resize();
+
+    // Handle optional parameters appropriately
+    let idPrefix: string;
+    if (this.cardArray.cluster.length > 0) {
+      if (this.cardArray.clusterIdPrefix) {
+        idPrefix = this.cardArray.clusterIdPrefix;
+      }
+      this.cardArray.cluster.forEach(function(card, index) {
+        // Set unique IDs for each card, if not already explicitely defined.
+        if (!card.id) {
+          card.id = idPrefix ? `${idPrefix}${index}` : `CardCluster${index + 1}`;
+        }
+      });
+    }
```

Comment:
what do you think about getting rid of this `idPrefix` variable? We could do it all by just checking `this.cardArray.clusterIdPrefix`? Although is a bit more of a mouthful 🤔 

```suggestion
    if (this.cardArray.cluster.length > 0) {
      this.cardArray.cluster.forEach(function(card, index) {
        // Set unique IDs for each card, if not already explicitely defined.
        if (!card.id) {
          card.id = this.cardArray.clusterIdPrefix ? `${this.cardArray.clusterIdPrefix}${index}` : `CardCluster${index + 1}`;
        }
      });
    }
```

Reply:
Accepting this in my next commit, since I had others I needed to do locally anyhow. 👍